keras/src/backend/mlx/core.py

```diff
@@ -71,8 +71,6 @@ def convert_to_tensor(x, dtype=None, sparse=None):
         return x.value
 
     if isinstance(x, np.ndarray):
-        if x.dtype == np.int64:
-            x = x.astype(np.int32)
         x = x.astype(standardize_dtype(x.dtype))
```

Comment:
I have deleted the above line as it casts numpy arrays of  int64 to int32 - let me know if I have to revert this one.

Reply:
Removing these lines are causing other test cases to fail. I will look into it tomorrow

Reply:
Isn't this because MLX doesn't support int64?

Reply:
https://ml-explore.github.io/mlx/build/html/python/data_types.html this page lists  int64 as supported dtype. One thing which is different is i32 is default integer type. It does not support float64.

Reply:
Numpy support for bf16 is through some extension ? If bf16 numpy array is converted to mlx numpy array without passing dtype  param - mlx is returning an array of dtype complex64 !! i don't know how to handle this.

Reply:
I have reverted this change for now. We can merge this pr if it is good enough.